test/parallel/test-setproctitle.js

```diff
@@ -8,37 +8,68 @@ if (common.isSunOS) {
 }
 
 const assert = require('assert');
-const exec = require('child_process').exec;
+const { exec, spawn } = require('child_process');
 const path = require('path');
-
 // The title shouldn't be too long; libuv's uv_set_process_title() out of
 // security considerations no longer overwrites envp, only argv, so the
-// maximum title length is possibly quite short.
-let title = String(process.pid);
-
-assert.notStrictEqual(process.title, title);
-process.title = title;
-assert.strictEqual(process.title, title);
+// maximum title length is possibly quite short (probably just 4 chars).
+// We need to use a random shared secret so zombie processes do not cause a
+// false positive.
+// Ref: https://github.com/nodejs/node/pull/12792
+const title = String(Date.now()).substr(-4, 4);
 
-// Test setting the title but do not try to run `ps` on Windows.
-if (common.isWindows) {
-  return common.skip('Windows does not have "ps" utility');
+if (process.argv[2] === 'child') {
+  const childTitle = process.argv[3];
+  assert.notStrictEqual(process.title, childTitle);
+  process.title = childTitle;
+  assert.strictEqual(process.title, childTitle);
+  // Just wait a little bit before exiting.
+  setTimeout(() => {}, common.platformTimeout(3000));
+  return;
 }
 
-// To pass this test on alpine, since Busybox `ps` does not
-// support `-p` switch, use `ps -o` and `grep` instead.
-const cmd = common.isLinux ?
-            `ps -o pid,args | grep '${process.pid} ${title}' | grep -v grep` :
-            `ps -p ${process.pid} -o args=`;
+// This should run only in the parent.
+assert.strictEqual(process.argv.length, 2);
+let child;
+const cmd = (() => {
+  if (common.isWindows) {
+    // On Windows it is the `window` that gets the title and not the actual
+    // process, so we create a new `window` by using `shell: true` and `start`.
+    child = spawn(
+      'start /MIN /WAIT',
+      [process.execPath, __filename, 'child', title],
+      { shell: true, stdio: 'ignore' },
+    );
+    const PSL = 'Powershell -NoProfile -ExecutionPolicy Unrestricted -Command';
+    const winTitle = '$_.mainWindowTitle';
+    return `${PSL} "ps | ? {${winTitle} -eq '${title}'} | % {${winTitle}}"`;
```

Comment:
I think this is subject to race condition, where the PowerShell script may run before the spawned child process changes the title? Maybe we can use `fork()` instead of `spawn()` and use the IPC channel for sequencing? Would also have the benefit of not having to kill the child.

Reply:
Can't use IPC because the script is run not in the child but in a grandchild created with `START`...

Reply:
Hm, is there any solution to this? Because without this solved it seems like there can not be any progress.